app-text/apvlv/apvlv-0.4.0.ebuild

```
@@ -0,0 +1,41 @@
+# Copyright 2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit xdg cmake desktop
+
+DESCRIPTION="Alf's PDF/DJVU/EPUB Viewer like Vim"
+HOMEPAGE="https://github.com/naihe2010/apvlv/"
+SRC_URI="https://github.com/naihe2010/apvlv/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="debug djvu"
+
+DEPEND="
+	>=app-text/poppler-0.5.0[cairo,xpdf-headers(+)]
+	dev-libs/glib:2
+	x11-libs/gtk+:3
+	net-libs/webkit-gtk
+	app-text/ebook-tools
+	media-libs/freetype
+	djvu? ( app-text/djvu:= )
+"
+src_configure() {
+	local mycmakeargs=(
+		-DAPVLV_WITH_DJVU=$(usex djvu)
+		-DAPVLV_ENABLE_DEBUG=$(usex debug)
+		-DAPVLV_WITH_TXT=ON
+	)
+	cmake_src_configure
+}
+src_install() {
+	cmake_src_install
+	newicon -s 32 icons/pdf.png ${PN}.png
+}
+RDEPEND="${DEPEND}"
+BDEPEND="
+	virtual/pkgconfig
+"
```

Let's keep the default ebuild structure here, where dependencies and overall the global stuff comes before ebuild phases.
```suggestion
RDEPEND="
	app-text/ebook-tools
	>=app-text/poppler-0.5.0:=[cairo,xpdf-headers(+)]
	dev-libs/glib:2
	dev-libs/libxml2
	net-libs/webkit-gtk
	x11-libs/cairo
	x11-libs/gdk-pixbuf:2
	x11-libs/gtk+:3
	x11-libs/pango
	djvu? ( app-text/djvu:= )
"
DEPEND="${RDEPEND}
	app-text/ghostscript-gpl"
BDEPEND="virtual/pkgconfig"
"

src_configure() {
	local mycmakeargs=(
		-DAPVLV_WITH_DJVU=$(usex djvu)
		-DAPVLV_ENABLE_DEBUG=$(usex debug)
		-DAPVLV_WITH_TXT=ON
	)
	cmake_src_configure
}

src_install() {
	cmake_src_install
	newicon -s 32 icons/pdf.png ${PN}.png
}

```
I also noticed the freetype dep seems bogus, even though upstream claims so. Maybe only headers are needed (add to DEPEND-only)? Also ghostscript-gpl seems to be necessary to get this installed.

---

created a patch for the CXX flags and fixed the rest